lib/huginn_agent.rb

```diff
@@ -1,17 +1,45 @@
 require 'huginn_agent/version'
 
 class HuginnAgent
-  def self.register(*paths)
-    @paths ||= []
-    paths.each do |path|
-      @paths << path
+  class << self
+    attr_accessor :branch, :remote
+
+    def load_tasks(options = {})
+      @branch = options[:branch] || 'master'
+      @remote = options[:remote] || 'https://github.com/cantino/huginn.git'
+      Rake.add_rakelib File.join(File.expand_path('../', __FILE__), 'tasks')
+    end
+
+    def load(*paths)
+      paths.each do |path|
+        load_paths << path
+      end
+    end
+
+    def register(*paths)
```

Comment:
The Agents need to inherit from `Agent` (the AR base model), so to allow this kind of registration `HuginnAgent` would need to be a subclass of `Agent` which is not defined yet when the gem is loaded into Huginn. Maybe I am missing something but due to the load order I think that would require us to monkey path (predefine) the Agent class before its defined in Huginn.
What I like about the approach that @cantino took is that the Agent code does not need any changes to be move into a gem.